### ISSUE TITLE
M2cryoto error Linux

### ISSUE THREAD

**Issue by corruptedkernel** [EmpireProject/Empire#1223]
Hey guys, I’m trying to figure out why Empire isn’t working for my kali Linux machine. I have both python2, and python3 installed, I did the “pip install -r requirements” no problems and did the install.sh. 

After install it’s giving me the M2crypto module can’t be found error, and then when trying to do “pip2 install m2crypto” it says ya ok and good to go and same goes for “pip3 install m2crypto” 

I’ve also tried deleting the entire empire folder and downloading from git again, and nothing.

Anyone have any ideas.

**Comment by True-Demon** [comment]
I ran into this earlier this morning when setting up Empire on fedora for the first time. You are likely missing other dependencies related to m2crypto and swig. You can try manually installing the packages with your local package manager if pip is unable to compile them.

I resolved the issue by installing them manually to my system like so (dnf for fedora, but same deal for debian-like systems)  
`sudo apt-get install m2crypto swig`

rerun `sudo pip2 install -r requirements.txt` and rerun the ./install.sh script as root and attempt to restart empire.

**Comment by kapshure** [comment]
> I ran into this earlier this morning when setting up Empire on fedora for the first time. You are likely missing other dependencies related to m2crypto and swig. You can try manually installing the packages with your local package manager if pip is unable to compile them.
> 
> I resolved the issue by installing them manually to my system like so (dnf for fedora, but same deal for debian-like systems)
> `sudo apt-get install m2crypto swig`
> 
> rerun `sudo pip2 install -r requirements.txt` and rerun the ./install.sh script as root and attempt to restart empire.

Hmmm that's the thing, the system already shows these as being installed : 
python-m2crypto is already the newest version (0.27.0-6).
swig is already the newest version (3.0.12-1.2).

and then running pip2 install -r requirements results in every package stated as "requirement already satisfied"

**Comment by mr64bit** [comment]
Can you try running `setup/install.sh` from the dev branch? Some fixes for kali have been added there.

**Comment by kapshure** [comment]
Had this working, haven't used it since mid-Oct, came back today to launch Empire, and back to the same message about "m2crypto" missing. Have tried everything already in this issue listing: dev branch vs master, apt installs of the package, pip install on requirements, reran setup, reran db setup, etc.   

am on kali 2019.1
4.19.0-kali1-amd64 #1 SMP Debian 4.19.13-1kali1 (2019-01-03) x86_64 GNU/Linux

ideas?

**Comment by mr64bit** [comment]
@kapshure Is this a fresh Kali image? I just tested on 2019.1 and everything went smoothly. Can you show me the output of `setup/install.sh`?

**Comment by kapshure** [comment]
no, and after I posted that last comment, I seem to have recalled having to go deploy another kali VM to make that work.  Let me try that before asking for further assistance, and/or spinning unnecessary cycles.

**Comment by truesamurai** [comment]
I have the same issue on my Kali. I tried on a new install of Kali and then there is no issue like this.
